api/src/routes/donate.ts

```diff
@@ -76,5 +87,141 @@ export const donateRoutes: FastifyPluginCallbackTypebox = (
     }
   );
 
+  fastify.post(
+    '/donate/charge-stripe-card',
+    {
+      schema: schemas.chargeStripeCard
+    },
+    async (req, reply) => {
+      try {
+        const { paymentMethodId, amount, duration } = req.body;
+        const { id } = req.session.user;
+
+        const user = await fastify.prisma.user.findUnique({
+          where: { id }
+        });
```

Comment:
```suggestion
        const user = await fastify.prisma.user.findUniqueOrThrow({
          where: { id }
        });
```
That way you can drop the `!user` check.